src/HDF5.jl

```diff
@@ -535,25 +536,33 @@ heuristic_chunk(x) = Int[]
 
 ### High-level interface ###
 # Open or create an HDF5 file
-function h5open(filename::AbstractString, rd::Bool, wr::Bool, cr::Bool, tr::Bool, ff::Bool,
+function h5open(filename::AbstractString, mode::String,
```

Comment:
The manual is pretty clear that I should use AbstractString "If you define a function expecting a string argument, you should declare the type as AbstractString in order to accept any string type." https://docs.julialang.org/en/stable/manual/strings/